docs/drill/_index.md

```diff
@@ -0,0 +1,23 @@
+---
+title: "Drill"
+bookIconImage: ../img/drill-logo.png
```

Comment:
Does this image need to be added as well?

Reply:
@rdblue I wasn't sure where exactly to add the image.  I submitted a PR for the Iceberg doc site [1] with the image.  Is that the wrong place?

[1]: https://github.com/apache/iceberg-docs/pull/62

Reply:
Looks good to me. Thanks!

Reply:
@rdblue Are we waiting for something to merge this?  Thanks!